lib/Doctrine/ODM/PHPCR/UnitOfWork.php

```diff
@@ -543,6 +543,10 @@ public function getOrCreateDocuments($className, $nodes, array &$hints = array()
                 $documentState[$class->parentMapping] = $this->getOrCreateProxyFromNode($node->getParent(), $locale);
             }
 
+            if ($class->depthMapping) {
+                $documentState[$class->depthMapping] = $node->getDepth();
```

Comment:
maybe simply add it to a document used in https://github.com/doctrine/phpcr-odm/blob/master/tests/Doctrine/Tests/ODM/PHPCR/Functional/BasicCrudTest.php ?